src/print_event_cb.c

```diff
@@ -550,8 +560,72 @@ print_header_bin(int argc, char *const argv[])
 	header.argc = argc;
 	data_size += offsetof(struct header_s, argv);
 
+	/* get current working directory */
 	char cwd[PATH_MAX];
 	if (getcwd(cwd, PATH_MAX) == NULL) {
+		perror("getcwd");
+		return -1;
+	}
+
+	/* get hardware identifier (utsname.machine) */
+	struct utsname buf;
+	if (uname(&buf)) {
+		perror("uname");
+		return -1;
+	}
+
+	/* currently only x86_64 is supported */
+	unsigned architecture;
+	if (strncmp(buf.machine, STR_ARCH_x86_64, strlen(STR_ARCH_x86_64)) == 0)
+		architecture = ARCH_x86_64;
+	else
+		architecture = ARCH_UNKNOWN;
+
+	/*
+	 *               *** FORMAT OF VLTRACE BINARY LOG ***
+	 *
+	 *	* first 80036 bytes is the same for all logs:
+	 *
+	 *	char [12]		- syscall table signature "VLTRACE_TAB"
+	 *	unsigned		- major version number
+	 *	unsigned		- minor version number
+	 *	unsigned		- patch version number
+	 *	unsigned		- hardware architecture
+	 *	struct sc_desc [1000]	- syscall table
+	 *
+	 *	* offset 80036 bytes is here
+	 *
+	 *	char [12]		- log signature "VLTRACE_LOG"
+	 *	int 			- BUF_SIZE
+	 *	int			- length of CWD (cwd_len)
+	 *	char [cwd_len]		- current working directory
+	 *	int			- size of the following struct header
+	 *	struct header		- structure containing the command line
+	 *
+	 *	* Here follows actual log consisting of pairs:
+	 *
+	 *	unsigned		- size of the packet
+	 *	struct data_*		- data packet
+	 */
+
+	/* save syscall table signature */
+	char tab_signature[] = VLTRACE_TAB_SIGNATURE;
+	if (fwrite(tab_signature, sizeof(tab_signature), 1, OutputFile) != 1) {
```

Comment:
Please verify at compile time that sizeof(tab_signature) is really 12, like you specified above.